apis/Google.Devtools.AspNet/Google.Devtools.AspNet.Tests/CloudTrace/LabelsTest.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Moq;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using System.Web;
+using Xunit;
+
+namespace Google.Devtools.AspNet.Tests.CloudTrace
+{
+    public class LabelsTest
+    {
+        [Fact]
+        public void FromHttpRequestWrapper()
+        {
+            HttpRequest request = new HttpRequest("name", "https://google.com", "");
+            Mock<HttpRequestWrapper> mockWrapper = new Mock<HttpRequestWrapper>(request);
```

Comment:
Why not just call `new HttpRequestWrapper(request)` and make sure the request has the information we need? I think it's cleaner not to use mocks unless we care about the interaction side (or have to for another reason).